src/core/bitmath_func.hpp

```diff
@@ -383,7 +383,7 @@ static inline T ROR(const T x, const uint8 n)
 	{
 #if !defined(__ICC) && defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4)  && __GNUC_MINOR__ >= 3))
 		/* GCC >= 4.3 provides a builtin, resulting in faster code */
-		return (uint32)__builtin_bswap32((int32)x);
+		return static_cast<uint32>(__builtin_bswap32(static_cast<int32>(x)));
```

Comment:
According to https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html, the cast to int32 is unnecessary (or possibly wrong)